misk/src/main/kotlin/misk/hibernate/ReflectionQueryFactory.kt

```diff
@@ -59,27 +53,13 @@ internal class ReflectionQuery<T : DbEntity<T>>(
     return typedQuery.list()
   }
 
-  override fun <P : Projection> listAs(session: Session, projection: KClass<P>): List<P> {
-    val criteriaBuilder = session.hibernateSession.criteriaBuilder
-    val query = criteriaBuilder.createQuery(Any::class.java)
-    val root: Root<*> = query.from(rootEntityType.java)
-
-    @Suppress("UNCHECKED_CAST") // The cache always returns matching types.
-    val projectionHandler = factory.projectionCache[projection] as ProjectionHandler<P>
-
-    query.select(projectionHandler.select(criteriaBuilder, root))
-    query.where(buildWherePredicate(root, criteriaBuilder))
-    val rows = session.hibernateSession.createQuery(query).list()
-    return rows.map { projectionHandler.toValue(it) }
-  }
-
   override fun invoke(proxy: Any, method: Method, args: Array<out Any>?): Any? {
-    val constraintHandler = queryMethodHandlers[method]
+    val handler = queryMethodHandlers[method]
 
     return when {
-      constraintHandler != null -> {
-        constraintHandler.invoke(this, args ?: arrayOf())
-        proxy
+      handler != null -> {
+        val result = handler.invoke(this, args ?: arrayOf())
```

Comment:
in practice, the return type here is either a Query or the final type (eg. List<Something>, SomeProjection, etc.), right?

Reply:
exactly